lms/djangoapps/courseware/model_data.py

```diff
@@ -784,7 +786,18 @@ def _fields_to_cache(self, blocks):
         for block in blocks:
             for field in block.fields.values():
                 scope_map[field.scope].add(field)
-        return scope_map
+
+        try:
+            block = blocks[0]
```

Comment:
Because the main actor here is not the XBlock itself but the asides that will be rendered alongside as those are not part of the course, but are generated dynamically.

Reply:
isn't it problematic to cache the `user_state_summary` fields of the aside xblock?

Reply:
Not sure, but I think this cache is per request, so every time a user requests the course content or calls an XBlock handler, the cache is regenerated.

This cache is read-write thought so being able to read the content will allow to write to it

Reply:
I see so you're getting all the asides from the first block, but can't asides choose what blocks they're applied to?  Is it possible that some asides would not be available on the first block but would be available on later blocks?

Reply:
I can do some local testing to verify that. It's true that asides choose on which blocks to render, but I'm not sure if this workflow applies to the function `get_aside_from_xblock`.

It seems this function always returns an aside xblock instance:

https://github.com/eduNEXT/edx-platform/blob/09a652379276551ede6070a33a7a15a65e674d4b/xmodule/modulestore/split_mongo/caching_descriptor_system.py#L420-L422

https://github.com/eduNEXT/edx-platform/blob/09a652379276551ede6070a33a7a15a65e674d4b/xmodule/modulestore/split_mongo/caching_descriptor_system.py#L403-L407